agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -3531,16 +3442,71 @@ backhaul_manager::create_link_metrics_collector(const sLinkInterface &link_inter
-    sMacAddr neighbor = network_utils::mac_from_string(controller_bridge_mac);
-    if ((neighbor_mac_filter == network_utils::ZERO_MAC) || (neighbor_mac_filter == neighbor)) {
-        neighbor_links_map[m_sConfig.wire_iface].push_back(neighbor);
+    // controller is not a neighbor) connected through the wired backhaul interface
+
+    sLinkInterface wired_interface;
+    wired_interface.iface_name = m_sConfig.wire_iface;
+
+    if (!get_media_type(wired_interface.iface_name, ieee1905_1::eMediaTypeGroup::IEEE_802_3,
+                        wired_interface.media_type)) {
+        LOG(ERROR) << "Unable to compute media type for interface " << wired_interface.iface_name;
+        return false;
+    }
+
+    if (!get_iface_mac(wired_interface.iface_name, wired_interface.iface_mac)) {
+        return false;
+    }
+
+    sLinkNeighbor neighbor;
+    neighbor.iface_mac = network_utils::mac_from_string(controller_bridge_mac);
+    neighbor.al_mac    = neighbor.iface_mac;
+    if ((neighbor_mac_filter == network_utils::ZERO_MAC) ||
+        (neighbor_mac_filter == neighbor.al_mac)) {
+        neighbor_links_map[wired_interface].push_back(neighbor);
+    }
+
+    // Also include a link for each associated client
+    for (const auto &slave : slaves_sockets) {
+
+        for (const auto &associated_clients_entry : slave->associated_clients_map) {
+            auto bssid              = associated_clients_entry.first;
+            auto associated_clients = associated_clients_entry.second;
+
+            sLinkInterface interface;
+            if (get_iface_name(bssid, interface.iface_name)) {
+                interface.iface_mac = bssid;
+                interface.media_type =
+                    beerocks::get_802_11_media_type(slave->freq_type, slave->max_bandwidth);
+
+                if (ieee1905_1::eMediaType::UNKNONWN_MEDIA == interface.media_type) {
+                    LOG(ERROR) << "Unknown media type for interface " << interface.iface_name;
+                    return false;
+                }
+            } else {
+                LOG(ERROR) << "Unable to get interface name for BSSID " << bssid;
+                return false;
+            }
+
+            LOG(TRACE) << "Getting neighbors connected to interface " << interface.iface_name
+                       << " with BSSID " << bssid;
+
+            for (const auto &associated_client : associated_clients) {
+                // TODO: This is not correct... We actually have to get this from the topology
+                // discovery message, which will give us the neighbor interface and AL MAC addresses.
+                neighbor.iface_mac = associated_client.first;
+                neighbor.al_mac    = neighbor.iface_mac;
+
+                if ((neighbor_mac_filter == network_utils::ZERO_MAC) ||
+                    (neighbor_mac_filter == neighbor.al_mac)) {
+                    neighbor_links_map[interface].push_back(neighbor);
+                }
+            }
+        }
```

Comment:
Since we're building up interface per interface, we don't actually need a map for neighbor_links_map. It could just as well be a vector-of-vectors. Of course, to make this a little bit manageable, that should be factored into structures or even better, classes.

But let's not do that for this PR.